lm_eval/generation.py

```diff
@@ -79,7 +79,17 @@ def parallel_generations(task, dataset, accelerator, model, tokenizer, n_tasks,
     # do not confuse args.batch_size, which is actually the num_return_sequences
     ds_loader = DataLoader(ds_tokenized, batch_size=1)
 
-    model, ds_loader = accelerator.prepare(model, ds_loader)
+    device = accelerator.device
+    if args.precision == "bf16":
+        print("Converting model to bf16")
+        model = model.bfloat16().to(device)
+    elif args.precision == "fp16":
+        print("Converting model to fp16")
+        model = model.half().to(device)
+    else:
+        model = model.to(device)
+
+    ds_loader = accelerator.prepare(ds_loader)
```

Comment:
Nice - is this just as good as directly passing `torch_dtype=` when loading the model?

Reply:
good point, probably cleaner to load the model directly in the specified precision instead of converting it

Reply:
updated to use `torch_dtype`